wagyu/cli/ethereum.rs

```diff
@@ -613,22 +630,30 @@ impl CLI for EthereumCLI {
     fn print(options: Self::Options) -> Result<(), CLIError> {
         fn output<N: EthereumNetwork, W: EthereumWordlist>(options: EthereumOptions) -> Result<(), CLIError> {
             let wallets = match options.subcommand.as_ref().map(String::as_str) {
-                Some("hd") => {
-                    let path = options.to_derivation_path(true).unwrap();
-                    (0..options.count)
-                        .flat_map(|_| {
-                            match EthereumWallet::new_hd::<N, W, _>(
-                                &mut StdRng::from_entropy(),
-                                options.word_count,
-                                options.password.as_ref().map(String::as_str),
+                Some("hd") => (0..options.count)
+                    .flat_map(|_| {
+                        let mut rnd = StdRng::from_entropy();
+                        let mnemonic = EthereumMnemonic::<N, W>::new_with_count(&mut rnd, options.word_count).unwrap();
+                        let password = options.password.as_ref().map(String::as_str);
+                        let mut opt = options.clone();
+                        let mut i = 0;
+
+                        (0..options.address_count).flat_map(move |_| {
```

Comment:
I believe you can use the variable currently ignored `|_|` as the iterator `i` so you don't need to instantiate a separate iterator for this mapping